-----------------------------------------------------------------
Revision: 13fbe0815f84d5b3c001947559f5818c10275f4c
Ancestor: cfe0e649dda34d9252d40d8f67e445336a247998
Author: nosnilmot@pidgin.im
Date: 05/17/2010 01:11:01 AM
Branch: im.pidgin.pidgin

Modified files:
        pidgin/gtkblist.c

ChangeLog: 

Prevent a null pointer dereference in chat_account_filter_func() if called
on a disconnected or disconnecting account. 
This should fix https://bugzilla.redhat.com/show_bug.cgi?id=592750


============================================================
--- pidgin/gtkblist.c	59572bc5415328dec42852173febb8e6d216f911
+++ pidgin/gtkblist.c	cbe664829ff51113e1fda2ec2c6f1791a3b46c47
@@ -942,6 +942,9 @@ chat_account_filter_func(PurpleAccount *
 	PurpleConnection *gc = purple_account_get_connection(account);
 	PurplePluginProtocolInfo *prpl_info = NULL;
 
+	if (gc == NULL)
+		return FALSE;
+
 	prpl_info = PURPLE_PLUGIN_PROTOCOL_INFO(gc->prpl);
 
 	return (prpl_info->chat_info != NULL);
